### PR TITLE
Bring various changes from Pro

### PR DESCRIPTION
**Pull request by ricardofandrade**
This PR has a number of items that seem to have been missed or purposely not brought into open source. The goal here is to determine what should be part of open source.

### CODE REVIEW

File: src/cpp/server/extras/admin/rstudio-server.in [hunk 1]
```diff
@@ -72,42 +71,45 @@ case "$1" in
         if [ -n "$2" ]; then
            kill -s USR1 $3 $4 $5 $6 $7 $8 $9 $2
         else
-           echo "Must specify PID of session to suspend"
+           argType="PID"
```
Comment (ricardofandrade) [inline src/cpp/server/extras/admin/rstudio-server.in:74]: Minimizing differences with Pro. Many occurrences.

File: src/cpp/core/include/core/system/UserObfuscation.hpp [hunk 2]
```diff
@@ -13,6 +13,20 @@
  *
  */
 
+// these macros obfuscate user IDs, with the goal of making it difficult to
```
Comment (kfeinauer) [inline src/cpp/core/include/core/system/UserObfuscation.hpp:16]: None of these changes should come to open source for this file.

File: src/cpp/session/SessionClientInit.cpp [hunk 3]
```diff
@@ -88,6 +88,15 @@ namespace session {
 namespace client_init {
 namespace {
 
+std::string userIdentityDisplay(const http::Request& request)
+{
+   std::string userIdentity = request.headerValue(kRStudioUserIdentityDisplay);
+   if (!userIdentity.empty())
+      return userIdentity;
+   else
+      return session::options().userIdentity();
```
Comment (ricardofandrade) [inline src/cpp/session/SessionClientInit.cpp:97]: We could always return this value instead.

File: src/cpp/session/include/session/SessionConstants.hpp [hunk 4]
```diff
@@ -85,7 +85,7 @@
 
 #define kUseSecureCookiesSessionOption    "session-use-secure-cookies"
 #define kIFrameEmbeddingSessionOption     "session-iframe-embedding"
-#define kLegacyCookiesSessionOption       "legacy-cookies"
+#define kLegacyCookiesSessionOption       "session-legacy-cookies"
```
Comment (ricardofandrade) [inline src/cpp/session/include/session/SessionConstants.hpp:88]: I missed this one myself.

File: src/gwt/src/org/rstudio/studio/client/application/Application.java [hunk 5]
```diff
@@ -222,8 +222,19 @@ public void onError(ServerError error)
...
+               // redirect to the specified URL (as a sub URL of the site's root)
+               String redirectUrl = ApplicationUtils.getHostPageBaseURLWithoutContext(false) + 
+            		   error.getRedirectUrl();
+               navigateWindowWithDelay(redirectUrl);
+            }
+            else
+            {
+               globalDisplay_.showErrorMessage("RStudio Initialization Error",
+                                               error.getUserMessage());
+            }
```
Comment (ricardofandrade) [inline src/gwt/src/org/rstudio/studio/client/application/Application.java:237]: There's an else here in Pro.

**Comment by gtritchie** [comment]
We generally don’t backport from Pro to Open Source. Changes need to flow from open source to Pro, see the section at end of https://github.com/rstudio/rstudio-pro/wiki/Pro-and-Open-Source.

**Comment by gtritchie** [comment]
Is purpose of this PR to cleanup such drift that has occurred and get things back in sync? All in favor of that!

**Comment by ricardofandrade** [comment]
Yup, that's the goal @gtritchie !
Some may be intentional, some not. We're in this PR to determine that.

File: src/cpp/session/modules/environment/SessionEnvironment.cpp [hunk 6]
```diff
@@ -1291,7 +1291,7 @@ Error initialize()
       (bind(registerRpcMethod, "remove_objects", removeObjects))
       (bind(registerRpcMethod, "remove_all_objects", removeAllObjects))
       (bind(registerRpcMethod, "get_environment_state", getEnv))
-      (bind(registerRpcMethod, "get_object_contents", getObjectContents))
+      (bind(registerRpcMethod, "getObjectect_contents", getObjectContents))
```
Comment (ricardofandrade) [inline src/cpp/session/modules/environment/SessionEnvironment.cpp:1294]: Is this really intentional?

Comment (kfeinauer) [reply src/cpp/session/modules/environment/SessionEnvironment.cpp:1294]: Definitely doesn't look like it - I'd blame it and discuss with the owner.

Comment (ricardofandrade) [reply src/cpp/session/modules/environment/SessionEnvironment.cpp:1294]: @MariaSemple it feels like an accidental search replace done in Pro.
The matching code seems to be:
https://github.com/rstudio/rstudio/blob/3de59e3c7fead55d3792d724847af3e21fadb96a/src/gwt/src/org/rstudio/studio/client/server/remote/RemoteServer.java#L6512

Comment (ricardofandrade) [reply src/cpp/session/modules/environment/SessionEnvironment.cpp]: I'll bring a fix to Pro myself if confirmed 😄

File: src/cpp/core/markdown/sundown/html_blocks.h [hunk 7]
```diff
@@ -2,7 +2,7 @@
 /* Command-line: gperf -N find_block_tag -H hash_block_tag -C -c -E --ignore-case html_block_names.txt  */
 /* Computed positions: -k'1-2' */
 
-#if !((' ' == 32) && ('!' == 33) && ('"' == 34) && ('#' == 35) \
+#if (!(' ' == 32) && ('!' == 33) && ('"' == 34) && ('#' == 35) \
```
Comment (kevinushey) [inline src/cpp/core/markdown/sundown/html_blocks.h:5]: Is this change intentional? It looks odd -- it seems to me the intention really is to negate the whole expression and not just the first piece.

File: src/cpp/core/zlib/zutil.h [hunk 8]
```diff
@@ -22,7 +22,7 @@
 #include "zlib.h"
 
 #if defined(STDC) && !defined(Z_SOLO)
-#  if !(defined(_WIN32_WCE) && defined(_MSC_VER))
+#  if (!defined(_WIN32_WCE) && defined(_MSC_VER))
```
Comment (kevinushey) [inline src/cpp/core/zlib/zutil.h:25]: This also seems odd?

File: src/cpp/monitor/include/monitor/events/Event.hpp [hunk 9]
```diff
@@ -46,48 +47,46 @@ enum EventScope
 #define kSessionAdminSuspend     2006
 #define kSessionAdminTerminate   2007
 
-class Event
+#define kMaxEventDataSize     32
```
Comment (kevinushey) [inline src/cpp/monitor/include/monitor/events/Event.hpp:50]: Can we please document with a comment why we choose 32 here? (If I understand correctly this is because it's the maximum username length on Linux.)

File: src/cpp/monitor/events/Event.cpp [hunk 10]
```diff
@@ -24,6 +24,24 @@
...
+             const std::string& username,
+             PidType pid,
+             boost::posix_time::ptime timestamp)
+   : empty_(false),
+     scope_(scope),
+     id_(id),
+     pid_(pid),
+     timestamp_(core::date_time::millisecondsSinceEpoch(timestamp))
+{
+   ::memset(&username_, 0, kMaxEventDataSize+1);
```
Comment (kevinushey) [inline src/cpp/monitor/events/Event.cpp:39]: Should we check and warn if we're truncating the username / data here? I also worry that we would fail to write the null terminator in these cases.

File: src/cpp/session/modules/environment/SessionEnvironment.cpp (hunk 6 above)
Comment (MariaSemple) [reply src/cpp/session/modules/environment/SessionEnvironment.cpp]: Yes, it looks like an accidental search and replace to me!

File: src/cpp/core/zlib/zutil.h (hunk 8 above)
Comment (ricardofandrade) [reply src/cpp/core/zlib/zutil.h:25]: @MariaSemple this feels like another accidental change since this file is not ours.
Thoughts?

File: src/cpp/core/markdown/sundown/html_blocks.h (hunk 7 above)
Comment (ricardofandrade) [reply src/cpp/core/markdown/sundown/html_blocks.h:5]: @MariaSemple another file we don't own.

File: src/cpp/core/zlib/zutil.h (hunk 8 above)
Comment (MariaSemple) [reply src/cpp/core/zlib/zutil.h:25]: Yes, I doubt I meant to do this.

File: src/cpp/core/markdown/sundown/html_blocks.h (hunk 7 above)
Comment (MariaSemple) [reply src/cpp/core/markdown/sundown/html_blocks.h]: Looks like it was also accidental, probably part of the same search/replace that changed the other `#if` block

File: src/cpp/monitor/events/Event.cpp (hunk 10 above)
Comment (ricardofandrade) [reply src/cpp/monitor/events/Event.cpp:39]: Events carry very little data. Digits or username only.

Comment (kevinushey) [reply src/cpp/monitor/events/Event.cpp:39]: I still think it's worth validating this is true, since IIUC we do construct these events from somewhat arbitrary inputs; e.g. in `eventFromJson()`:

https://github.com/rstudio/rstudio-pro/blob/4caacd464bb703d5741863fea7b6a530ca7ad045/src/cpp/monitor/http/Server.cpp#L118-L134

https://github.com/rstudio/rstudio-pro/blob/4caacd464bb703d5741863fea7b6a530ca7ad045/src/cpp/monitor/events/EventUtils.cpp#L33-L58

Comment (ricardofandrade) [reply src/cpp/monitor/events/Event.cpp:40]: Warn log added
